extension/packagemanagerextension/nbextension/scripts.js

```diff
@@ -0,0 +1,132 @@
+
+define([
+    'jquery',
+    './views',
+    './api',
+    './common'
+], function ($, views, api, common) {
+    "use strict";
+
+
+    /*
+    This function populates the sidebar each time it is opened.
+    */
+
+    async function packageview(dir) {
+
+        let info = await api.getinfo(dir);
+        let data = info.packages;
+
+        let output = views.installed(data);
+        $('#installed-packages').html(output);
+
+        output = views.toinstall(data, []);
+        $('#to-install').html(output);
+        if (output === "") {
+            $('#to-install-main').css("display", "none");
+        }
+
+        let selectedPackages = [];
+        views.selectinstalled(selectedPackages);
+
+        let toInstall = [];
+        views.selecttoinstall(toInstall);
+    }
+
+
+    /*
+    This function executes the given function after the specified timeout.
+    */
+
+    function delay(fn, ms) {
+        let timer = 0;
+        return function (...args) {
+            clearTimeout(timer);
+            timer = setTimeout(fn.bind(this, ...args), ms || 0);
+        };
+    }
+
+    /*
+    This function populates the dropdown datalist when something is searched.
+    */
+
+    function searchview() {
+
+        $(function () {
+            $('#package-name').unbind();
+            $('#package-name').keyup(delay(async function (e) {
+                let query = this.value;
+                $('#searchicon').toggleClass('fa-search fa-spinner');
+                let res = await api.search(query);
+                let pks = res.packages;
+                let html = ""
+                $('#searchlist').html(html);
+                Array.from(pks).forEach(element => {
+                    let name = element.name;
+                    let version = element.version;
+                    let entry = name + " - " + version;
+                    html += "<option value='" + entry + "'>";
+                    html += entry;
+                    html += "</option>";
+                });
+                $('#searchicon').toggleClass('fa-search fa-spinner');
+                $('#searchlist').html(html);
+
+                $('input[list="searchlist"]').each(function () {
+                    var elem = $(this),
+                        list = $(document.getElementById(elem.attr('datalist')));
+                    elem[0].value = elem[0].value.split('=')[0];
+                    elem.autocomplete({
+                        source: list.children().map(function () {
+                            return $(this).text();
+                        }).get()
+                    });
+                });
+
+            }, 1000));
+        });
+
+        document.querySelector('input[list="searchlist"]').addEventListener('input', onInput);
```

Comment:
why not using jquery to be consistent with the rest of the code??? I've said this thing how many times???